File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarHrData.java
// Copyright © 2019 Polar Electro Oy. All rights reserved.
package com.polar.sdk.api.model;

import java.util.ArrayList;
import java.util.List;

/**
 * For heart rate data.
 */
public class PolarHrData {

    /**
     * Heart rate in BPM (beats per minute).
     */
    public final int hr;

    /**
     * R is the peak of the QRS complex in the ECG wave and RR is the interval between successive Rs.
     * In 1/1024 format.
     */
    public final List<Integer> rrs;

    /**
     * RRs in milliseconds.
     */
    public final List<Integer> rrsMs;

    /**
     * Equals true if the sensor has contact (with a measurable surface e.g. skin).
     */
    public final boolean contactStatus;

    /**
     * Equals true if the sensor supports contact status
     */
    public final boolean contactStatusSupported;

    /**
     * Equals true if RR data is available.
     */
    public final boolean rrAvailable;

    public PolarHrData(int hr, List<Integer> rrs, boolean contactStatus, boolean contactStatusSupported, boolean rrAvailable) {
        this.hr = hr;
        this.rrs = rrs;
        this.contactStatus = contactStatus;
        this.contactStatusSupported = contactStatusSupported;
        this.rrAvailable = rrAvailable;
        rrsMs = new ArrayList<>();
        for (int rrRaw : rrs) {
            rrsMs.add((int) (Math.round(((float) rrRaw / 1024.0) * 1000.0)));
        }
    }
}

File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarHrData.kt
// Copyright © 2019 Polar Electro Oy. All rights reserved.
package com.polar.sdk.api.model

data class PolarHrData(
    val samples: List<PolarHrSample>
) {
    /**
     * Polar heart rate sample
     * @property hr value is heart rate in BPM (beats per minute).
     * @property ppgQuality PPG signal quality of the real time HR between 0 and 100
     * @property correctedHr Corrected value of the real time HR value. 0 if unavailable.
     * @property rrsMs RRs in milliseconds. R is a the top highest peak in the QRS complex of the ECG wave and RR is the interval between successive Rs.
     * @property rrAvailable true if RR data is available.
     * @property contactStatus true if the sensor has contact (with a measurable surface e.g. skin)
     * @property contactStatusSupported  true if the sensor supports contact status
     */
    data class PolarHrSample(
        val hr: Int,
        val correctedHr: Int,
        val ppgQuality: Int,
        val rrsMs: List<Int>,
        val rrAvailable: Boolean,
        val contactStatus: Boolean,
        val contactStatusSupported: Boolean,
    )
}